controller/engine_controller.go

```diff
@@ -562,6 +564,7 @@ func (ec *EngineController) isMonitoring(e *longhorn.Engine) bool {
 func (ec *EngineController) startMonitoring(e *longhorn.Engine) {
 	stopCh := make(chan struct{})
 	monitor := &EngineMonitor{
+		logger:           ec.logger.WithField("engine", e.Name),
```

Comment:
Would `engine-monitor` be better?

Reply:
The object we are concerned with is a `engine` not the monitor.

Reply:
Example, log print from below to make it more clear what it mean by that.
`log.Warnf("need to wait for the restore error handling before the restore invocation: %v", status.Error)`
this is concerned with a `engine` not a monitor, the monitor itself is only an internal construction that is not exposed to the outside.

Reply:
Why is the `engine` preferred?

Reply:
See comment above?
We want to correlate actions across components that are involved with the `engine` object.

Maybe this helps?
It wouldn't make much sense calling it `engine-monitor` here then in the longhorn-engine/instance-manager project the field would be called `engine`?

Reply:
Let me flip the question, why do you think `engine-monitor` would be better?

Reply:
Got it! Sorry for missing the comment, since the page was not refreshed when I replied to you.

Reply:
I was thinking the original logger is based on the controller, hence this derived logger should base on the monitor.